packages/store/src/ivy/ivy-enabled-in-dev-mode.ts

```diff
@@ -21,9 +21,8 @@ export function setIvyEnabledInDevMode(): void {
   try {
     // `try-catch` will also handle server-side rendering, as
     // `window is not defined` will not be thrown.
-    const ng = (window as any).ng;
-    const _ivyEnabledInDevMode =
-      !!ng && typeof ng.getComponent === 'function' && typeof ng.markDirty === 'function';
+    const ng = window as any;
+    const _ivyEnabledInDevMode = !!ng && typeof ng.getComponent === 'function';
```

Comment:
View engine exposes `ng.probe` and `ng.coreTokens` in all modes, Ivy doesn't. What do you think if we ran our condition against view engine properties:
```ts
const ng = window as any;
const _viewEngineEnabled = !!ng && !!ng.probe || !!ng.coreTokens;
// `isDevMode()` can be taken from the InjectionToken
const _ivyEnabledInDevMode = !_viewEngineEnabled && isDevMode();
```
Thus we won't care about those functions like `markDirty`, `getComponent` etc. Because Angular team removed `markDirty` without mentioning it in changelog, that's why we had this regression.